core-bundle/src/Cron/MessengerCron.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCronJob;
+use Contao\CoreBundle\Util\ProcessUtil;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Promise\Utils;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+use Symfony\Component\Process\Process;
+
+#[AsCronJob('minutely')]
+class MessengerCron
+{
+    /**
+     * @var callable|null
+     */
+    private $promiseFactory;
+
+    /**
+     * @param array<array{'options': array<string>, 'transports': array<string>, 'autoscale': array{'enabled': bool, 'desired_size': int, 'max': int}}> $workers
+     */
+    public function __construct(private ContainerInterface $messengerTransportLocator, private string $consolePath, private array $workers)
+    {
+    }
+
+    public function __invoke(string $scope): PromiseInterface|null
+    {
+        if (Cron::SCOPE_CLI !== $scope) {
+            return null;
+        }
+
+        $workerPromises = [];
+
+        foreach ($this->workers as $worker) {
+            $this->addWorkerPromises($worker, $workerPromises);
+        }
+
+        return Utils::all($workerPromises);
+    }
+
+    public function setPromiseFactory(callable $callable): self
```

Comment:
Is this method only used for testing?
If so, would it make sense to refactor `ProcessUtil` into a service so that it can be mocked instead and this method not needed anymore then?

Reply:
Yes. None of our `Util` classes are currently a service. Do you think I can just make it one anyway?

Reply:
I don’t know if we would have to rename it to something not Util then. But a service seams to be a better fit here I think.

Reply:
Converted in https://github.com/contao/contao/pull/5405/commits/6d24eda65108ea84ab72987028a32df7b3889788.